Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +57,34 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool voltageCheck = true;
+    bool currentCheck = true;
+
     QStringList sections = line.split(BATDATA_DELIMITER);
 
     QString timeString = sections.at(0);
+
+    if (sections.size() != CORRECT_SECTION_SIZE)
+    {
+        return false;
+    }
+
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    batteryData.voltage = sections.at(1).toDouble(&voltageCheck);
 
-    batteryData.current = sections.at(2).toDouble();
+    batteryData.current = sections.at(2).toDouble(&currentCheck);
 
-    return true;
 
+    if (!batteryData.time.isValid())
+    {
+        return false;
+    }
+
+    if (voltageCheck == true && currentCheck == true )
```

Comment:
This if statement can be combined with the previous one. Also, use the `if(boolean)` format for better readability.